src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types/Person';
+import { getPeople } from '../api';
+import { Error, NoPeopleMessage } from '../components/Error';
+import { useSortAndSearch } from '../hooks/useSortAndSearch';
+import { getSearchWith } from '../utils/searchHelper';
+import { SortSex } from '../types/SortSex';
+import { Sort } from '../types/Sort';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+  const { personSlug = '' } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || SortSex.all;
+  const query = searchParams.get('query') || '';
+  const sort = searchParams.get('sort') || Sort.none;
+  const order = searchParams.get('order') || '';
+
+  async function fetchPeople() {
+    setIsError(false);
+    try {
+      setIsLoading(true);
+      const peopleData = await getPeople();
+
+      setPeople(peopleData);
+      setIsLoading(false);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  useEffect(() => {
+    fetchPeople();
+  }, []);
+
+  const sortedAndSearchedPeople = useSortAndSearch(
+    people,
+    sex,
+    centuries,
+    query,
+    sort,
+  );
+
+  if (order) {
```

Comment:
It's better to use a useMemo hook to avoid unnecessary reversions on every render.